app/controllers/addresses_controller.rb

```diff
@@ -0,0 +1,48 @@
+class AddressesController < ApplicationController
+  before_action :authenticate_user!
+  before_action :redirect_edit_if_present, only: :new
```

Comment:
action内に直接書いてもいいけどね(まあこれでも良い)

Reply:
actionがfatになるのが嫌だったのでこうしました。
名前変えた方がいいかもですね
```suggestion
  before_action :redirect_edit_if_address_present, only: :new
```